### PR TITLE
Stage collision and some player_1.s

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/game/multiplayer/player_unk_1.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #define GUARD_MULTIPLAYER_PLAYER_UNK_1_H
 
 #include "global.h"
-#include "core.h"
+#include "core.h" // for MultiSioData
```
Comment (freshollie) [inline include/game/multiplayer/player_unk_1.h:7]: At some point we move this out of core. It's unrelated

File: src/game/bosses/boss_9.c [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+{
+    TA53Boss *boss;
+    TA53_unk1C *unk1C;
+    TA53_unk48 *unk48;
+    u8 i;
+
+    gActiveBossTask = TaskCreate(Task_EggmanKidnapsVanilla, sizeof(TA53Boss), 0x4000, 0,
+                                 TaskDestructor_TrueArea53BossGfx);
+
+    CreateTrueArea53Boss();
```
Comment (freshollie) [inline src/game/bosses/boss_9.c:96]: Does this just init the boss then if it's already been allocated in this func?

File: src/game/stage_collision.c [hunk 3]
```diff
@@ -0,0 +1,891 @@
...
+
+    return r3;
+}
+
+// TODO: Fix this register mess!
+// (100.00%) https://decomp.me/scratch/xGy3C
+s32 sub_801EE64(s32 p0in, s32 p1in, s32 p2in, u8 *p3in)
+{
+#ifndef NON_MATCHING
+    register u32 r0 asm("r0");
```
Comment (freshollie) [inline src/game/stage_collision.c:586]: lol

File: src/game/stage_collision.c [hunk 4]
```diff
@@ -0,0 +1,891 @@
...
+    register s32 r3 asm("r3");
+    register s32 p0 asm("r4") = p0in;
+    register s32 p1 asm("r5") = p1in;
+    register s32 r6 asm("r6");
+    register s32 p2 asm("r8") = p2in;
+    u32 r7;
+    register s32 sb asm("sb");
+    register u8 *p3 asm("sl") = p3in;
+    register s32 res asm("r4");
+    register u8 *hm asm("r1");
```
Comment (freshollie) [inline src/game/stage_collision.c:597]: Lol I'm surprised this worked

File: src/game/player_0.c [hunk 5]
```diff
@@ -1889,3 +1897,480 @@ NONMATCH("asm/non_matching/game/sub_8022D6C.inc", void sub_8022D6C(Player *p))
...
+            }
+
+            result = fnRes;
+        } break;
+    }
+
+    return result;
+}
+
+// (98.07%) https://decomp.me/scratch/xgjsf
```
Comment (freshollie) [inline src/game/player_0.c:2007]: May take a look at this later

**Review by freshollie — APPROVED** [review]
Very chunky. If you wanna merge now, feel free

File: src/game/bosses/boss_9.c (hunk 2 above)
Comment (JaceCear) [reply src/game/bosses/boss_9.c:96]: Well, it's named like all other boss creation functions that are referenced in `game/bosses/common.c`'s array.
And both `CreateTrueArea53Boss` and this function allocate the same number of bytes with a `TaskCreate` call each.

File: include/game/multiplayer/player_unk_1.h (hunk 1 above)
Comment (JaceCear) [reply include/game/multiplayer/player_unk_1.h:7]: Sounds good!

File: src/game/bosses/boss_9.c (hunk 2 above)
Comment (freshollie) [reply src/game/bosses/boss_9.c:96]: Right, so `SetupEggmanKidnapsVanillaTask` actually ends up creating 2 instances of the boss? Cos it allocates 1 here, and then creates another task in `CreateTrueArea53Boss`?

**Pull request merged**